config/functions

```diff
@@ -484,34 +484,57 @@ save_build_config() {
   done
 }
 
-check_config() {
-  dashes="==========================="
-  if [ ! -d $PROJECT_DIR/$PROJECT ]; then
-    check_project="$check_project\n $dashes$dashes$dashes"
-    check_project="$check_project\n ERROR: Project not found, use a valid project or create a new config"
-    check_project="$check_project\n $dashes$dashes$dashes"
-    check_project="$check_project\n\n Valid projects:"
-
-    for projects in $PROJECT_DIR/*; do
-      check_project="$check_project\n - $(basename $projects)"
+check_distro() {
+  local dashes="===========================" distro_err_msg
+  if [ -z "${DISTRO}" -o ! -d "${DISTRO_DIR}/${DISTRO}" ]; then
+    distro_err_msg=\n $dashes$dashes$dashes"
+    distro_err_msg=$distro_err_msg\n ERROR: Distro not found, use a valid distro or create a new config"
+    distro_err_msg=$distro_err_msg\n $dashes$dashes$dashes"
+    distro_err_msg=$distro_err_msg\n\n Valid distros:"
+
+    for distros in ${DISTRO_DIR}/*; do
+      distro_err_msg="$distro_err_msg\n - ${distros##*/}"
     done
-    echo -e $check_project
+    echo -e $distro_err_msg
```

Comment:
If you add `-e` support to die, then can't you combine this so that it becomes `die "$distro_err_msg"`?